src/callback_queue.h

```diff
@@ -0,0 +1,64 @@
+#ifndef SRC_CALLBACK_QUEUE_H_
+#define SRC_CALLBACK_QUEUE_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include <atomic>
+
+namespace node {
+
+template <typename R, typename... Args>
```

Comment:
While in here, it may be helpful to add some code comments that would help future contributors find their way around. Like, for instance, describing what is R expected to be here...

Reply:
Sure, done! :+1: